inductiva/fluids/scenarios/wind_terrain/terrain.py

```diff
@@ -0,0 +1,147 @@
+"""Represent a complex terrain profile."""
+import typing
+import random
+
+import pyvista as pv
+
+from inductiva.generative import procedural
+
+
+class Terrain:
+    """Represents a Terrain profile.
+    
+    A Terrain profile is defined by the surface of the terrain.
+    Here, we describe it by a mesh with coordinates (x, y, z).
+    The (x, y) reflect the (longitude, latitude) of the terrain
+    and the z coordinate is the elevation of the terrain.
+    """
+
+    def __init__(self, terrain_mesh):
+        """Initializes the `Terrain` object.
+        
+        Args:
+            terrain_mesh (pyvista.StructuredGrid): Terrain mesh."""
+        self.mesh = terrain_mesh
+
+    @classmethod
+    def from_file(cls, terrain_file: str):
+        """Creates a `Terrain` object from a text file."""
+        terrain_mesh = pv.read(terrain_file)
+
+        return cls(terrain_mesh)
+
+    @classmethod
+    def from_random_generation(cls,
+                               x_range: typing.Sequence[float],
+                               y_range: typing.Sequence[float],
+                               x_num: int,
+                               y_num: int,
+                               height_factor: float = 10,
+                               initial_roughness: float = 1,
+                               roughness_factor: float = 0.5):
+        """Creates a `Terrain` object with random elevations.
+        The elevation of the corners are randomly selected within
+        the range of 0 to 1. This limits the maximum terrain height
+        due to the nature of the terrain generation algorithm.
+        To allow for more complex terrains, we allow the user to multiply
+        by a `heigh_factor` to increase the maximum height of the terrain.
+        In this way, the maximum_height possible will be around 
+        the value of the `height_factor`.
+
+        TODO: Improve terrain generation for high discrepancy
+        terrains.
+
+        Args:
+            x_range = [x_min, x_max]: Range of the x-axis in meters (m).
+            y_range = [y_min, y_max]: Range of the y-axis in meters (m).
+            x_num: Number of points in the x-axis.
+            y_num: Number of points in the y-axis.
+            height_factor: Factor to multiply the elevation by.
+            initial_roughness: Initial roughness of the terrain.
+            roughness_factor: Factor to multiply the roughness by.
+        """
+        corner_values = [
+            random.uniform(0, 1),
+            random.uniform(0, 1),
+            random.uniform(0, 1),
+            random.uniform(0, 1)
+        ]
+
+        x_grid, y_grid, z_elevation = procedural.generate_random_terrain(
```

Comment:
OK! This is why I think we need to refactor this. 
There is one aspect which is getting a m_pixel by n_pixel grid. This grid does not depend on the final physical dimension, it just gives you a random z map of  m x n dimension. The physical "dimensions" are computed after by scaling. I think these should be two separate methods. Let me look again at the code that was submitted before.